app/routes/docs.client.samples.md

```diff
@@ -0,0 +1,160 @@
+---
+handle:
+  breadcrumb: Code Samples
+---
+
+# Code Samples
+
+We have collected a short list of functions and examples that may be useful
+when working with GCN Notices in your code. These include conversions from
+text and VOEvent alert formats to JSON and XML (for VOEvent alerts), how to save alerts,
+and some samples from the FAQs section of the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
+
+For more examples, or to contribute your own ideas, see our [Utility Samples](https://github.com/nasa-gcn/gcn.nasa.gov/blob/CodeSamples/app/routes/docs.client.samples.md) and the [gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python) repository.
+
+## Parsing
+
+Within your consumer loop, use the following functions to convert the
+content of <code>message.value()</code> into other data types.
+
+<code>xmltodict</code> is not a standard python package, it must be installed with
+pip:
+
+```sh
+pip install xmltodict
+```
+
+```python
+
+import email
+import xml.etree.ElementTree as ET
+import xmltodict
+
+def parse_text_alert_to_dict(message_value):
+    return dict(email.message_from_bytes(message_value))
+
+def parse_voevent_alert_to_xml_root(message_value):
+    return ET.fromstring(message_value)
+
+def parse_voevent_alert_to_dict(message_value):
+    return xmltodict.parse(message_value)
+
+```
+
+## Saving
+
+Use the following to save the data to a local file:
+
+```python
+import json
+
+def save_alert(message):
+    with open('path/to/your/file', 'wb') as file:
+        file.write(message.value())
+
+```
+
+## Keep track of the last read message when restarting a client
+
+A key feature of kafka consumer clients is the ability to perform persistent
+tracking of which messages have been read. This allows clients to recover
+missed messages after a restart by beginning at the earliest unread message
+rather than the next available message from the stream.
```

Comment:
```suggestion
Kafka can keep track of which messages your client has read, allowing your client to recover
missed messages after a restart by beginning at the earliest unread message
rather than the next available message from the stream.
```